### ISSUE TITLE
/uplift-vagrant/crm-dev vagrant sometimes fails

### ISSUE THREAD

**Issue by shurick81** [SubPointSolutions/uplift-contrib#5]
vagrant output: https://gist.github.com/shurick81/392d1ef6330cb0762a041b99581d865e
dynamics installation log: https://gist.github.com/shurick81/b618fbbe2bcdb1293df50688a8b05b1e

**Comment by shurick81** [comment]
This is the error in the log:
```
Error| Check PrivReportingGroupValidator : Failure: The group is not valid.
The server is not operational.
    CN=CRM01PrivReportingGroup,OU=CRM groups,DC=uplift,DC=local
```

**Comment by shurick81** [comment]
A workaround is to re-run:
```
vagrant destroy --force
vagrant up
```

**Comment by shurick81** [comment]
I connected via RDP and run this in PowerShell:
```
[ADSI]"LDAP://CN=CRM01PrivReportingGroup,OU=CRM groups,DC=uplift,DC=local"
```
it returned an object so no issue here.

**Comment by shurick81** [comment]
Found a support article but diagnostics is positive, no issues:
https://support.microsoft.com/en-ie/help/4045916

```
C:\Windows\system32>echo %logonserver%
\\DC

C:\Windows\system32>nltest /dsgetdc:uplift.local
           DC: \\dc.uplift.local
      Address: \\192.168.16.5
     Dom Guid: 3044a258-552a-4153-b1ec-b8fa877b1b33
     Dom Name: uplift.local
  Forest Name: uplift.local
 Dc Site Name: Default-First-Site-Name
Our Site Name: Default-First-Site-Name
        Flags: PDC GC DS LDAP KDC TIMESERV GTIMESERV WRITABLE DNS_DC DNS_DOMAIN DNS_FOREST CLOSE_SITE FULL_SECRET WS DS_8 DS_9 DS_10
The command completed successfully
```

**Comment by avishnyakov** [comment]
Right, from what we see, it might be that groups or other AD objects aren't ready yet?

CRM provision consists of several steps, one of which setups a minimal set of AD users and groups. At least two ideas here:
1) re-ensure that AD objects are there

We actually do this for DC provision. Some of the win services take time to get up, so we try/catch until they come online and only then report a successful DC setup. Bypassing this results in failing user/group resolution against AD. Could be same case here.

2) Enhance Dynamics365Configuration setup, use updated version on this module

All modules are pinned, we pin versions to avoid configuration drifts. As you can see, current output does not provide CRM intall progress:
```
crm: UPLIFT : 04-06-2019 20:14:04.006 : INFO    : CRM/vagrant : Skipping -LogFilePullToOutput option, v1.0.0.0 module or older
```

We need to update Dynamics365Configuration version here:

https://github.com/SubPointSolutions/uplift-contrib/blob/master/uplift-vagrant/crm-dev/scripts-crm/crm-ps-modules.ps1#L14

You already merged Dynamics365Configuration PR with this feature, right?
[options for real-time installation feedback and log file location #22](https://github.com/shurick81/Dynamics365Configuration/pull/22)

That way we would see real-time output. Logs won't get lost during first Vagrant provision.

So for the group, after DSC group setup, we might do it several times, try to lookup the group until it's there.

One more option - is there any way to run CRM pre-install validation before actually install?
